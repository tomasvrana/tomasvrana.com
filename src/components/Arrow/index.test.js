import React from 'react'
import Arrow from './'
import { shallow } from 'enzyme'

function getWrapper (props) {
  return shallow(
    <Arrow {...props} />
  )
}

describe('<Arrow />', () => {
  test('it renders correctly with no props', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  test('it renders an "up" arrow correctly', () => {
    expect(getWrapper({ up: true })).toMatchSnapshot()
  })

  test('it renders a "right" arrow correctly', () => {
    expect(getWrapper({ right: true })).toMatchSnapshot()
  })

  test('it renders a "down" arrow correctly', () => {
    expect(getWrapper({ down: true })).toMatchSnapshot()
  })

  test('it renders a "left" arrow correctly', () => {
    expect(getWrapper({ left: true })).toMatchSnapshot()
  })

  test('it renders a "filled" arrow correctly', () => {
    expect(getWrapper({ fill: true })).toMatchSnapshot()
  })
})
