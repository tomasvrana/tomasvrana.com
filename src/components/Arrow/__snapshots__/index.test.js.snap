// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> it renders a "down" arrow correctly 1`] = `
<styled.div
  rotation={90}
>
  <IoMdPlay />
</styled.div>
`;

exports[`<Arrow /> it renders a "filled" arrow correctly 1`] = `
<styled.div
  rotation={90}
  shouldFill={true}
>
  <IoMdPlay />
</styled.div>
`;

exports[`<Arrow /> it renders a "left" arrow correctly 1`] = `
<styled.div
  rotation={180}
>
  <IoMdPlay />
</styled.div>
`;

exports[`<Arrow /> it renders a "right" arrow correctly 1`] = `
<styled.div
  rotation={0}
>
  <IoMdPlay />
</styled.div>
`;

exports[`<Arrow /> it renders an "up" arrow correctly 1`] = `
<styled.div
  rotation={270}
>
  <IoMdPlay />
</styled.div>
`;

exports[`<Arrow /> it renders correctly with no props 1`] = `
<styled.div
  rotation={90}
>
  <IoMdPlay />
</styled.div>
`;
